rosbag2_py/src/rosbag2_py/info_sorting_method.cpp

```diff
@@ -0,0 +1,103 @@
+// Copyright 2024 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "rosbag2_storage/bag_metadata.hpp"
+
+#include "info_sorting_method.hpp"
+
+
+namespace rosbag2_py
+{
+
+InfoSortingMethod info_sorting_method_from_string(std::string str)
+{
+  std::transform(str.begin(), str.end(), str.begin(), ::tolower);
+  auto find_result = sorting_method_map.find("str");
```

Comment:
Sorry, it seems I made a mistake originally putting `str` in quotes.
```suggestion
  auto find_result = sorting_method_map.find(str);
```
This is proof of why we need unit tests :wink: